ddtrace/contrib/internal/falcon/middleware.py

```diff
@@ -87,20 +83,13 @@ def process_response(self, req, resp, resource, req_succeeded=None):
 
         route = req.root_path or "" + req.uri_template
 
-        trace_utils.set_http_meta(
-            span,
-            config.falcon,
-            status_code=status,
-            response_headers=resp._headers,
-            route=route,
-        )
-
         # Emit span hook for this response
         # DEV: Emit before closing so they can overwrite `span.resource` if they want
         config.falcon.hooks.emit("request", span, req, resp)
 
-        # Close the span
-        span.finish()
```

Comment:
There's another `span.finish()` in `if resource is None:` - should that also be calling the core dispatch?

Reply:
Ive removed this finish call and moved it to use core.dispatch the same as the others.